src/components/GroupJoin/GroupGalleryCardsUI.vue

```diff
@@ -0,0 +1,257 @@
+      <div
+        v-if="showMyGroups && !previewOpened"
+        class="row no-wrap">
+        <div style="width: 100%; padding: 0">
+          <q-card style="margin-top: 16px">
+            <q-search
+              @change="$emit('search', arguments[0])"
+              class="searchbar"
+              v-model="search"
+            />
+          </q-card>
+        </div>
+        <div style="width: 100px">
+          <slot />
+        </div>
+      </div>
+    </transition>
+    <slot v-if="!showMyGroups"/>
+    <transition name="slide-toggle">
+      <q-alert
+        v-if="!isLoggedIn && expanded"
+        color="info"
+        icon="star"
+        class="alert"
+      >
+        <i18n path="JOINGROUP.LOGOUT_MESSAGE.LOGGED_OUT">
+          <router-link
+            place="login"
+            :to="{ name: 'login' }"
+            class="underline"
+          >
+            {{ $t('JOINGROUP.LOGOUT_MESSAGE.LOG_IN') }}
+          </router-link>
+        </i18n>
+      </q-alert>
+    </transition>
+    <transition name="slide-toggle">
+      <div
+        v-if="(expanded && filteredMyGroups.length>0) || (filteredOtherGroups.length == 0 && filteredMyGroups.length == 1)"
+        class="join-groups">
+        <h4 class="text-primary">
+          {{ $t('JOINGROUP.MY_GROUPS') }}
+        </h4>
+        <transition-group
+          name="list-complete"
+          v-if="showMyGroups"
+          class="row">
+          <div
+            v-for="group in filteredMyGroups"
+            :key="group.id"
+            class="list-complete-item inline-block col-xs-12 col-sm-6 col-lg-4 items-stretch"
+            :class="filteredMyGroups.length == 1 ? 'col-xs-12 col-sm-12 col-lg-12' : ''">
+            <GroupGalleryCard
+              v-if="filteredMyGroups.length != 1"
+              :class="{highlight: group.id === currentGroupId}"
+              :group="group"
+              :is-member="true"
+              @preview="showPreview(group)"
+              @visit="$emit('visit', { groupId: group.id })"
+            />
+            <GroupPreview
+              v-if="filteredMyGroups.length == 1"
+              :show-close="previewOpened"
+              @close="hidePreview()"
+              :group="group"
+              :is-logged-in="isLoggedIn"/>
+          </div>
+        </transition-group>
+      </div>
+    </transition>
+    <h4
+      class="text-primary generic-padding"
+      v-if="(expanded && filteredOtherGroups.length > 0) || !showMyGroups"
+    >
+      {{ $t('JOINGROUP.WHICHGROUP') }}
+    </h4>
+    <transition name="slide-toggle">
+      <q-card v-if="!previewOpened && !showMyGroups">
+        <q-search
+          @change="$emit('search', arguments[0])"
+          class="searchbar"
+          v-model="search"
+        />
+      </q-card>
+    </transition>
+    <transition name="slide-toggle">
+      <div v-if="(expanded && filteredOtherGroups.length > 0) || (filteredOtherGroups.length == 1 && filteredMyGroups.length == 0)">
```

Comment:
< insert comment about complicated logic in templates here > ;)

Reply:
Haha :laughing: